src/services/usersService.js

```diff
@@ -120,22 +126,31 @@ class UsersService {
   }
 
   async logout() {
-    navigator.serviceWorker.ready
-      .then((reg) =>
-        reg.pushManager.getSubscription().then((sub) =>
-          sub.unsubscribe().then(async () => {
-            await api.pushSubscriptionDelete();
-            await api.userLogout();
-            localStorage.removeItem("sessionId");
-          })
-        )
-      )
-      .catch(async (err) => {
-        console.error(err);
-        await api.userLogout();
-        localStorage.removeItem("sessionId");
-        throw new Error("User logout error");
-      });
+    const performLogoutRequest = async () => {
+      const response = await api.userLogout();
+
+      if (!response.success) {
```

Comment:
should it be `if (response.success) {`. ?

Reply:
No, this request will only work if the client loses the token and the server returns an invalid token error. In such a case, it is necessary to forcibly disconnect the user from the server.